src/backend/mssql.rs

```diff
@@ -0,0 +1,180 @@
+                Some(ref m) => format!(" DEFAULT '{}'", m),
+                _ => format!(""),
+            },
+            match tt.nullable {
+                true => "",
+                false => " NOT NULL",
+            },
+            match tt.unique {
+                true => " UNIQUE",
+                false => "",
+            },
+        )
+    }
+
+    fn drop_column(name: &str) -> String {
+        format!("DROP COLUMN [{}]", name)
+    }
+
+    fn rename_column(old: &str, new: &str) -> String {
+        format!(r#"EXEC sp_rename '{}', '{}'"#, old, new)
+    }
+
+    fn create_index(table: &str, schema: Option<&str>, name: &str, _type: &Type) -> String {
+        // FIXME: Implement PG specific index builder here
+        format!(
+            "CREATE {} INDEX [{}] ON {}[{}] ({})",
+            match _type.unique {
+                true => "UNIQUE",
+                false => "",
+            },
+            name,
+            prefix!(schema),
+            table,
+            match _type.inner {
+                BaseType::Index(ref cols) => cols
+                    .iter()
+                    .map(|col| format!("[{}]", col))
+                    .collect::<Vec<_>>()
+                    .join(", "),
+                _ => unreachable!(),
+            }
+        )
+    }
+
+    fn drop_index(name: &str) -> String {
+        format!("DROP INDEX [{}]", name)
+    }
+}
+
+impl MsSql {
+    fn prefix(ex: bool) -> String {
+        match ex {
+            true => format!("ADD "),
+            false => format!(""),
+        }
+    }
+
+    fn print_type(t: BaseType, schema: Option<&str>) -> String {
+        use self::BaseType::*;
+        match t {
+            Text => format!("TEXT"),
+            Varchar(l) => match l {
+                0 => format!("VARCHAR(MAX)"), // For "0" remove the limit
+                _ => format!("VARCHAR({})", l),
+            },
+            /* "NOT NULL" is added here because normally primary keys are implicitly not-null */
+            Primary => format!("IDENTITY(1,1) PRIMARY KEY NOT NULL"),
+            Integer => format!("INT"),
+            Float => format!("FLOAT(24)"),
+            Double => format!("FLOAT(53)"),
+            UUID => format!("UNIQUEIDENTIFIER"),
+            Boolean => format!("BIT"),
+            Date => format!("DATE"),
+            Json => format!("JSON"),
+            Binary => format!("VARBINARY(MAX)"),
+            Foreign(s, t, refs) => format!(
+                "INT REFERENCES {}[{}]({})",
+                quoted_prefix!(s.or(schema.map(|s| s.into()))),
+                t,
+                refs.0
+                    .iter()
+                    .map(|r| format!("[{}]", r))
+                    .collect::<Vec<_>>()
+                    .join(",")
+            ),
+            Custom(t) => format!("{}", t),
+            Array(meh) => format!("{}[]", MsSql::print_type(*meh, schema)),
+            Index(_) => unreachable!(), // Indices are handled via custom builder
```

Comment:
Maybe adding the note inside the `unreachable` statement is more descriptive if we ever hit this.
```suggestion
            Index(_) => unreachable!("Indices are handled via custom builder"),
```